catalog/core/catalog-core-directorymonitor/src/main/java/org/codice/ddf/catalog/content/monitor/ContentDirectoryMonitor.java

```diff
@@ -151,143 +162,134 @@ public void setAttributeOverrides(List<String> attributeOverrides) {
+                    "Unable to configure Camel route - this Content Directory Monitor will be unusable");
+        } finally {
             if (LOGGER.isDebugEnabled()) {
                 dumpCamelContext("after configureCamelRoute()");
             }
-        } catch (Exception e) {
-            LOGGER.warn(
-                    "Unable to configure Camel route - this Content Directory Monitor will be unusable",
-                    e);
         }
-
-        LOGGER.trace("EXITING: configureCamelRoute");
-    }
-
-    public List<RouteDefinition> getRouteDefinitions() {
-        return camelContext.getRouteDefinitions();
     }
 
-    private void startRoutes() {
-        LOGGER.trace("ENTERING: startRoutes");
-        List<RouteDefinition> routeDefinitions = camelContext.getRouteDefinitions();
-        for (RouteDefinition routeDef : routeDefinitions) {
-            startRoute(routeDef);
-        }
-        LOGGER.trace("EXITING: startRoutes");
+    /*
+        Assign this CDM's routeCollection to the routes generated by this CDM's RouteBuilder.
+     */
+    private void setRouteCollection(RouteBuilder routeBuilder) {
+        routeCollection = routeBuilder.getRouteCollection()
+                .getRoutes();
     }
 
-    private void startRoute(RouteDefinition routeDef) {
-        String routeId = routeDef.getId();
-        try {
-            if (isMyRoute(routeId)) {
-                ServiceStatus routeStatus = camelContext.getRouteStatus(routeId);
-                // Only start the route if it is not already started
-                if (routeStatus == null || !routeStatus.isStarted()) {
-                    LOGGER.trace("Starting route with ID = {}", routeId);
-                    camelContext.startRoute(routeDef);  //DEPRECATED
-                    // this method does not reliably start a route that was created, then
-                    // app shutdown, and restarted
-                    //                camelContext.startRoute(routeId);
-                }
+    /*
+        Do not attempt to add routes to the CamelContext until we know the content scheme is ready.
+     */
+    private void waitForContentCamelComponent() throws InterruptedException {
+        final int maxRetries = 20;
+        final int sleepSeconds = 5;
+        int retryCount = 0;
+        while (camelContext.getComponent("content") == null) {
+            TimeUnit.SECONDS.sleep(sleepSeconds);
+            retryCount++;
+            if (retryCount > maxRetries) {
+                throw new IllegalStateException("Could not get Camel component 'content' ");
             }
-        } catch (Exception e) {
-            LOGGER.info("Unable to start Camel route with route ID = {}", routeId, e);
         }
     }
 
+    /*
+        Only remove routes that this Content Directory Monitor created since the same CamelContext
+        is shared across all Content Directory Monitors.
+     */
     private void removeRoutes() {
         LOGGER.trace("ENTERING: stopRoutes");
```

Comment:
I personally clean up these trace methods for entering and exiting methods when I see them.